sqlx-core/src/error.rs

```diff
@@ -103,6 +103,16 @@ pub enum Error {
     #[cfg(feature = "migrate")]
     #[error("{0}")]
     Migrate(#[source] Box<crate::migrate::MigrateError>),
+
+    /// LTree labels can only contain [A-Za-z0-9_]
+    #[cfg(feature = "ltree")]
+    #[error("ltree label cotains invalid characters")]
+    InvalidLtreeLabel,
+
+    /// LTree version not supported
+    #[cfg(feature = "ltree")]
+    #[error("ltree version not supported")]
+    InvalidLtreeVersion,
```

Comment:
These errors are too specific for this enum.

I'd suggest creating a separate error type for the
`FromStr` impl of `Ltree` and then in decode return it boxed in `Error::Decode`.

Reply:
(We also often get lazy and just return strings since they can be converted to `Box<Error>`)